agent/agent.go

```diff
@@ -149,19 +166,101 @@ func newWithCommunicator(ctx context.Context, opts Options, comm client.Communic
 			Permissions: pail.S3PermissionsPublicRead,
 			ContentType: "text/plain",
 		}
+		opts.TraceCollectorEndpoint = setupData.TraceCollectorEndpoint
 	}
 
-	return &Agent{
+	a := &Agent{
 		opts:   opts,
 		comm:   comm,
 		jasper: jpm,
-	}, nil
+	}
+
+	a.closers = append(a.closers, closerOp{
+		name: "communicator close",
+		closerFn: func(ctx context.Context) error {
+			if a.comm != nil {
+				a.comm.Close()
+			}
+			return nil
+		}})
+
+	if err := a.initTracerProvider(ctx); err != nil {
+		return nil, errors.Wrap(err, "initializing tracer provider")
+	}
+
+	return a, nil
 }
 
-func (a *Agent) Close() {
-	if a.comm != nil {
-		a.comm.Close()
+func (a *Agent) initTracerProvider(ctx context.Context) error {
+	if a.opts.TraceCollectorEndpoint == "" {
+		return nil
 	}
+
+	resource, err := resource.New(ctx,
+		resource.WithAttributes(semconv.ServiceName("evergreen-agent")),
```

Comment:
This will work for now! Long term we might wanna think about how to make this something that's settable at the project level, maybe? 🤔

Reply:
It definitely seems possible, though since the agent is shared between projects I wonder if we'd want to divide the traces by project further downstream.
One potential issue that comes to mind is that HC doesn't allow queries across datasets and some stuff (like the host metrics) might straddle more than one project.

I'm good to wait and see.